rfcs/41-ui-navigation.md

```diff
@@ -0,0 +1,394 @@
+# Feature Name: Generic UI Navigation API
+
+## Summary
+
+A set of components and a new system to define navigation nodes for the ECS UI.
+
+## Motivation
+
+On top of making it easy to implement controller navigation of
+in-game UIs, it enables keyboard navigations and answers accessibility concerns
+such as the ones
+[raised in this issue](https://github.com/bevyengine/bevy/issues/254).
+
+## User-facing explanation
+
+We add a _navigation tree_ to the UI system, the tree will be used to enable
+key-based navigation of the specified UIs.
+
+Currently, the developer must themselves write the relationship between
+various buttons in the menus and specify which button press leads to other menu
+items, etc. Or more simply, limit themselves to pointer-based navigation to
+avoid having to think about menu navigation.
+
+By including navigation-related components such as `Navigable` and `Container`,
+the developer can quickly get a controller-navigable menu up and running. The
+`Container` accepts various `Plane`s to fine-tune more precisely which keys
```

Comment:
Users have no idea what either a container or a plane is here, unless they've already seen them in other UI frameworks.